topostats/tracing/disordered_tracing.py

```diff
@@ -0,0 +1,708 @@
+            a gaussian 'sigma' and number of dilation iterations.
+        skeletonisation_params : dict
+            Skeletonisation Parameters. Method of skeletonisation to use 'topostats' is the original TopoStats
+            method. Three methods from scikit-image are available 'zhang', 'lee' and 'thin'.
+        pruning_params : dict
+            Dictionary of pruning parameters. Contains 'method', 'max_length', 'height_threshold', 'method_values' and
+            'method_outlier'.
+        n_grain : int
+            Grain number being processed (only  used in logging).
+        """
+        self.image = image * 1e-9 if convert_nm_to_m else image
+        self.mask = mask
+        self.filename = filename
+        self.pixel_to_nm_scaling = pixel_to_nm_scaling * 1e-9 if convert_nm_to_m else pixel_to_nm_scaling
+        self.min_skeleton_size = min_skeleton_size
+        self.mask_smoothing_params = (
+            mask_smoothing_params
+            if mask_smoothing_params is not None
+            else {"gaussian_sigma": None, "dilation_iterations": 2}
+        )
+        self.skeletonisation_params = (
+            skeletonisation_params if skeletonisation_params is not None else {"method": "zhang"}
+        )
+        self.pruning_params = pruning_params if pruning_params is not None else {"method": "topostats"}
+        self.n_grain = n_grain
+        # Images
+        self.smoothed_mask = np.zeros_like(image)
+        self.skeleton = np.zeros_like(image)
+        self.pruned_skeleton = np.zeros_like(image)
+        # Trace
+        self.disordered_trace = None
+
+        # suppresses scipy splining warnings
+        warnings.filterwarnings("ignore")
+
+        LOGGER.debug(f"[{self.filename}] Performing Disordered Tracing")
+
+    def trace_dna(self):
+        """Perform the DNA skeletonisation and cleaning pipeline."""
+        # LOGGER.info(f"[{self.filename}] : mask_smooth_params : {self.mask_smoothing_params=}")
+        self.smoothed_mask = self.smooth_mask(self.mask, **self.mask_smoothing_params)
+        self.skeleton = getSkeleton(
+            self.smoothed_mask,
+            self.mask,
+            method=self.skeletonisation_params["method"],
+            height_bias=self.skeletonisation_params["height_bias"],
+        ).get_skeleton()
+        self.pruned_skeleton = prune_skeleton(self.smoothed_mask, self.skeleton, **self.pruning_params.copy())
+        self.pruned_skeleton = self.remove_touching_edge(self.pruned_skeleton)
+        self.disordered_trace = np.argwhere(self.pruned_skeleton == 1)
+
+        if self.disordered_trace is None:
+            LOGGER.info(f"[{self.filename}] : Grain failed to Skeletonise")
+        elif len(self.disordered_trace) < self.min_skeleton_size:
+            self.disordered_trace = None
+
+    def re_add_holes(
+        self, orig_mask: npt.NDArray, new_mask: npt.NDArray, holearea_min_max: list = (4, np.inf)
```

Comment:
I wonder if it would be more descriptive to use `smoothed_mask` here than `new_mask` as the method is applied to a smoother mask.

```suggestion
        self, orig_mask: npt.NDArray, smoothed_mask: npt.NDArray, holearea_min_max: list = (4, np.inf)
```

You should be able to easily change this across the code base using the [Rename symbol](https://code.visualstudio.com/docs/editor/refactoring#_rename-symbol) although in my experience it _doesn't_ make the necessary changes to the doc-string so you would have to manually update that field.

Reply:
The idea was to keep it as general as possible but if it's more readable this way then that's better!